src/ast/function-object.ts

```diff
@@ -28,23 +33,26 @@ module.exports = class FunctionObject {
         }
       }
     });
-
     // Now we analyze the body with the local context. Note that recursion is
     // allowed, because we've already inserted the function itself into the
     // outer context, so recursive calls will be properly resolved during the
     // usual "outward moving" scope search. Of course, if you declare a local
     // variable with the same name as the function inside the function, you'll
     // shadow it, which would probably be not a good idea.
     if (this.body) {
-      this.body.forEach(s => s.analyze(context));
+      this.body.forEach((s: Statement) => s.analyze(context));
     }
   }
 
-  optimize() {
-    this.parameters.forEach(p => p.optimize());
-    this.body.forEach(s => s.optimize());
-    this.body = this.body.filter(s => s !== null);
+  public optimize(): FunctionObject {
+    this.params.forEach((p: Parameter) => p.optimize());
+    this.body.forEach((s: Statement) => s.optimize());
+    this.body = this.body.filter((s: Statement) => s !== null);
     // Suggested: Look for returns in the middle of the body
     return this;
   }
-};
+
+  // Depends on the target language, thus gets filled in
+  // by the necessary generator at runtime.
+  public gen(): any { }
```

Comment:
Not consistent as sometimes there are `any`s mentioned (like here) and sometimes not.

Reply:
Good catch! Will resolve this in an upcoming commit !